custom_components/ble_monitor/ble_parser/xiaomi.py

```diff
@@ -204,6 +205,27 @@ def obj0007(xobj):
         return {}
     return {"door": door, "door action": action}
 
+def obj0008(xobj, device_type):
+    """armed away"""
+    returnData = {}
+    value = xobj[0] ^ 1
+    returnData.update({'armed away': value})
+    if len(xobj) == 5:
+        timestamp = int.from_bytes(xobj[1:], 'little')
+        timestamp = datetime.fromtimestamp(timestamp).isoformat()
+        returnData.update({'timestamp': timestamp})
+    # Lift up door handle outside the door sends this event from DSL-C08.
+    if device_type == "DSL-C08":
+        return{
+            "lock": value,
+            "locktype": 'lock',
+            "action": 'lock outside the door',
+            "method": "manual",
+            "error": None,
+            "key id": None,
+            "timestamp": None,
+        }
+    return returnData
```

Comment:
If I understand this correct, your lock ("DSL-C08") will always return the return function, never the `returnData`. Why did you than implement the `armed away` and `timestamp`? Is that a possible outcome for other sensors that aren't implemented yet (if yes, than it is OK, just curious).

Reply:
According to the Mi Home protocol, the disarm event is timestamped, but the disarm event of my lock is timestamped without.